src/rred_reports/reports/tables.py

```diff
@@ -0,0 +1,210 @@
+    """
+    Args:
+        school_filter(pd.DataFrame)
+
+    Returns:
+        table with the following columns
+            Number of RR teachers
+            Number of pupils served
+            (Pupil outcomes) Discontinued
+            (Pupil outcomes) Referred to school
+            (Pupil outcomes) Incomplete
+            (Pupil outcomes) Left School
+            (Pupil outcomes) Ongoining
+    """
+    columns_used = ["rred_user_id", "pupil_no", "exit_outcome"]
+
+    def filter_summary_table(school_df):
+        """Filter for summary table
+
+        Args: pd.DataFrame
+        """
+        return school_df
+
+    def get_outcome_from_summary(school_df, outcome_type):
+        try:
+            return school_df["exit_outcome"].value_counts()[outcome_type]
+        except KeyError:
+            return 0
+
+    filtered = filter_summary_table(school_df)
+    summary_table = filtered[columns_used]
+
+    return pd.DataFrame(
+        {
+            "number_of_rr_teachers": [summary_table["rred_user_id"].nunique()],
+            "number_of_pupils_served": [summary_table["pupil_no"].nunique()],
+            "po_discontinud": get_outcome_from_summary(summary_table, "Discontinued"),
+            "po_referred_to_school": get_outcome_from_summary(summary_table, "Referred to school"),
+            "po_incomplete": get_outcome_from_summary(summary_table, "Incomplete"),
+            "po_left_school": get_outcome_from_summary(summary_table, "Left school"),
+            "po_ongoing": get_outcome_from_summary(summary_table, "Ongoing"),
+        }
+    )
+
+
+def make_table(school_df: pd.DataFrame, template_path: Path):
+    """Function to fill the template with the dataframe
+
+    Args: pd.DataFrame
+    """
+
+    # adding a column for table four
+    school_df["total_lost_lessons"] = school_df[list(school_df.columns)[45:49]].sum(axis=1)
+
+    columns_and_filters = (
+        (table_one_columns, filter_one),
+        (table_two_columns, filter_two),
+        (table_three_columns, filter_three),
+        (table_four_columns, filter_four),
+        (table_five_columns, filter_five),
+        (table_six_columns, filter_six),
+    )
+
+    header_rows = [2, 1, 1, 1, 1, 2, 2]
+
+    # adding in summary table first
+    template_filler = TemplateFiller(template_path, header_rows)
+    add_in_summary_table = summary_table_function(school_df)
+    template_filler.populate_table(0, add_in_summary_table)
+    template_filler.save_document(r"output/reports/test.docx")
+
+    # now adding other tables
+    for index, column_and_filter in enumerate(columns_and_filters):
+        template_filler = TemplateFiller("output/reports/test.docx", header_rows)
+        columns, filter_function = column_and_filter
+        filtered = filter_function(school_df)
+        table_to_write = filtered[columns]
+        template_filler.populate_table(index + 1, table_to_write)
+        template_filler.save_document(r"output/reports/test.docx")

```

Comment:
I think we can reuse the same template filler object and only save it once
```suggestion
    template_filler = TemplateFiller(template_path, header_rows)
    add_in_summary_table = summary_table_function(school_df)
    template_filler.populate_table(0, add_in_summary_table)

    # now adding other tables
    for index, column_and_filter in enumerate(columns_and_filters):
        columns, filter_function = column_and_filter
        filtered = filter_function(school_df)
        table_to_write = filtered[columns]
        template_filler.populate_table(index + 1, table_to_write)

    template_filler.save_document(r"output/reports/test.docx")

```